### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by michael-mcgee**
Please review code for merge into master.

### CODE REVIEW

**Comment by mcrd25** [comment]
# Under review by TSE

File: lib/game.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    elsif @board.draw?
+      @board.display
+      puts "It's a draw!"
+      true
+    else
+      false
+    end
+  end
+
+  def play_again?
```
Comment (mcrd25) [inline lib/game.rb]: You can add this logic to mian.rb

File: lib/player.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  attr_reader :number, :name, :marker
+
+  def initialize(number, marker)
+    @number = number
+    @name = ''
+    @marker = marker
+  end
+
+  def prompt_for_name
+    print "Please enter name of player #{@number}: "
```
Comment (mcrd25) [inline lib/player.rb]: You can have this in main.rb and initialize the class with when you get it in main.rb

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+# Board class
+class Board
+  INITIAL_MARKER = ' '
+  WINNING_LINES = [
```
[the remark this answers is not in the document]
Comment (mcrd25) [reply lib/board.rb]: May be best to have a method calculate this dynamically if you need some tips, please feel free to contact me.
Or comment here.

**Comment by michael-mcgee** [comment]
Completed changes as requested. Please re-review.

**Comment by maelfosso** [comment]
# Under review by Mael FOSSO

File: bin/main.rb [hunk 4]
```diff
@@ -0,0 +1,49 @@
+# frozen_string_literal: true
+
+require '../lib/game'
+require '../lib/board'
+require '../lib/player'
```
Comment (maelfosso) [inline bin/main.rb]: I suggest you to use `require_relative`. I was obliged to change it to make it work.

File: bin/main.rb [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      puts "#{game.current_player.name}(#{game.current_player.marker}), it's your turn."
+      print 'Enter your selection (1-9): '
+      selection = gets.chomp.to_i
+      break if board.valid_selection?(selection) && (1..9).include?(selection)
+
+      puts 'Sorry, that selection is invalid!'
+    end
+
+    game.turn(selection)
+    break unless game.state == :active
```
Comment (maelfosso) [inline bin/main.rb]: `:active`, that's a good way to code

File: bin/main.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  system 'clear'
+  puts board.display
+
+  puts game.state == :winner ? "#{game.current_player.name} wins!" : "It's a draw!"
```
Comment (maelfosso) [inline bin/main.rb]: Nice code !!

File: lib/board.rb [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+    lines = []
+    # rows
+    lines += markers.each_slice(3).to_a
+    # cols
+    lines += markers.each_slice(3).to_a.transpose
+    # diagonals
+    lines << [markers[0], markers[4], markers[8]]
+    lines << [markers[2], markers[4], markers[6]]
+
+    lines.any? { |line| line.all? { |m| m == 'X' } || line.all? { |m| m == 'O' } }
```
Comment (maelfosso) [inline lib/board.rb]: Could you use `:X` syntax here?

File: bin/main.rb [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  end
+  player1 = Player.new(player_names[0], 'X')
+  player2 = Player.new(player_names[1], 'O')
+  game = Game.new(board, player1, player2)
+
+  # Player turn loop
+  loop do
+    game.switch_player
+    system 'clear'
+    puts board.display
```
Comment (maelfosso) [inline bin/main.rb]: `board` is an attribute of the `Game` class. Why are you using it like this? Why not `game.board` like it should be called?

File: bin/main.rb [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+
+    selection = ''
+    loop do
+      puts "#{game.current_player.name}(#{game.current_player.marker}), it's your turn."
+      print 'Enter your selection (1-9): '
+      selection = gets.chomp.to_i
+      break if board.valid_selection?(selection) && (1..9).include?(selection)
```
Comment (maelfosso) [inline bin/main.rb]: The same remark about `board`

**Comment by maelfosso** [comment]
Hello coders,
You guys have done a great job so far implementing this project! Before moving forward, there are some changes to do.

- [ ] Please, correct errors from **stickler-ci**

- [ ] Please, read the comment above

- [ ] You're supposed to use `game.board` not `board`. I indicated place up there.

Please, kindly correct those remarks and submit a new code review.

**Comment by AdnanAfsari** [comment]
# **Adnan Afsari working on it**

**Pull request merged**